paymentsheet/src/main/java/com/stripe/android/paymentsheet/SavedPaymentMethodMutator.kt

```diff
@@ -103,6 +108,26 @@ internal class SavedPaymentMethodMutator(
                 }
             }
         }
+
+        coroutineScope.launch {
+            canEdit.collect { canEdit ->
+                if (!canEdit && editing.value) {
+                    toggleEditing()
+                }
+            }
+        }
+
+        coroutineScope.launch {
+            paymentMethods.collect { paymentMethods ->
+                if (paymentMethods.isEmpty() && editing.value) {
+                    toggleEditing()
```

Comment:
I wonder if it'd be better to set `editing = false` here instead of toggling. If we had multiple state flows that all tried to toggle editing, it seems like we could end up with a race condition where it's toggled too many times

Reply:
Yeah, seems reasonable. I'll follow up.

Reply:
thanks!

Reply:
https://github.com/stripe/stripe-android/pull/8845